File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {
        int n = colors.length;
        int maxDistance = 0;
        for (int i = 0; i < n - 1; i++) {
            for (int j = n - 1; j > i + maxDistance; j--) {
                if (colors[i] != colors[j]) {
                    maxDistance = Math.max(maxDistance, j - i);
                    break;
                }
            }
        }
        return maxDistance;
    }
}

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        res = -1
        for i in range(len(colors) - 1):
            for j in range(len(colors) - 1, i, -1):
                if colors[i] != colors[j]:
                    res = max(res, j - i)
                    break
        return res